jax/_src/ad_checkpoint.py

```diff
@@ -224,7 +224,8 @@ def fun_remat(*args, **kwargs):
 ### Utilities
 
 def saved_residuals(f, *args, **kwargs) -> List[Tuple[core.AbstractValue, str]]:
-  args, in_tree = tree_flatten((args, kwargs))
+  args_seq, in_tree = tree_flatten((args, kwargs))
+  args = tuple(args_seq)
```

Comment:
I'm curious, was this change actually needed? It seems surprising. (It'd be cleaner not to add the extra line!)

Reply:
It was tripping a typing error on the commit hook.  We could get rid of the extra line by keeping
```python
args_seq, in_tree = tree_flatten((args, kwargs))
```
here, but then changing the two uses of it to `*args_seq` and `len(args_seq)` below.  Would that be preferable?  I was trying to be minimally invasive, but you're right that shorter is probably clearer.

Reply:
The reason that this is a type error by the way is because `args` changes type.  As an argument, it's a tuple, but the result of `tree_flatten` is either a sequence or a list—neither of which is a subclass of tuple.  To be honest, it might be better to give this a whole new name anyway since it doesn't correspond to just `args` anyway.  Maybe `in_leaves` to match `in_tree`?

Reply:
Ah, makes sense! `in_leaves` sgtm.